android/src/main/java/com/segment/analytics/kotlin/android/plugins/AndroidLifecyclePlugin.kt

```diff
@@ -260,14 +264,25 @@ class AndroidLifecyclePlugin() : Application.ActivityLifecycleCallbacks, Default
         application.unregisterActivityLifecycleCallbacks(this)
         if (useLifecycleObserver) {
             // only unregister if feature is enabled
-            lifecycle.removeObserver(this)
+            run(Runnable { lifecycle.removeObserver(this) })
+        }
+    }
+    private fun run(runnable: Runnable) {
```

Comment:
you don't have to use runnable in kotlin, try [high order function and lambda](https://kotlinlang.org/docs/lambdas.html)